bessctl/pcapngdump.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (c) 2017, Nefeli Networks, Inc.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+# list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# * Neither the names of the copyright holders nor the names of their
+# contributors may be used to endorse or promote products derived from this
+# software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import binascii
+import struct
+
+# Read from file `f` exactly `size` bytes.  Might return a buffer
+# smaller than `size` if EOF or error.
+def read_exactly(f, size):
+    ret = b''
+    while True:
+        ret += f.read(size)
+        size = size - len(ret)
+        if size == 0:
+            break
+    return ret
+
+# Return `a` rounded up to the nearest multiple of `b`
+def round_up_to_multiple(a, b):
+    return ((a + (b - 1)) / b) * b
+
+# Dump to file `outfile` the contents of the pcapng file `pcapng_file`.
+# Print this per packet info:
+# * An hex dump of the packet data
+# * The optional pcapng comment block as a string
+# If `count` > 0, return after reading at most `count` packets.
+def dump(pcapng_file, outfile, count):
+    shb = read_exactly(pcapng_file, 16)
+
+    if len(shb) < 16:
+        outfile.write("Unexpected EOF while reading SHB\n")
+        return
+
+    struct_endianness = '<'
+    block_type, bom = struct.unpack(struct_endianness + 'IxxxxIxxxx', shb)
+
+    if block_type != 0x0A0D0D0A:
```

Comment:
Could you define this and other the other magic numbers below as constants?

Reply:
Done, thanks